core/src/executor/execute.rs

```diff
@@ -245,9 +269,50 @@ async fn execute_inner<T: GStore + GStoreMut>(
             selection,
         } => {
             let columns = fetch_columns(storage, table_name).await?.map(Rc::from);
+            let referencings = storage.fetch_referencings(table_name).await?;
```

Comment:
now `delete` also look big enough to be splitted into a separate module as we do for `insert` and `update`. how about adding `delete` module and move these codes to that?